### PR TITLE
Person entity, CarDao, ComponentDaoTest

### PR DESCRIPTION
**Pull request by RobertTamas**

### CODE REVIEW

**Comment by RobertTamas** [comment]
Je to trosku hop alebo trop s dao operaciami :sweat_smile: takze budeme to musiet najskor vsetko spojit a potom si to budeme fixovat

File: src/main/test/cz/muni/fi/DaoTests/ComponentDaoTest.java [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    public void findComponentByID() {
+        Component foundComponent = componentManager.findComponent(engine3.getId());
+        Assert.assertEquals(foundComponent, engine3);
+    }
+
+    @Test
+    public void addDeleteComponent() {
+        Component component = new Component();
+        component.setName("Breaks");
+        component.setType(EComponentType.BREAKS);
```
Comment (richihrmo) [inline src/main/test/cz/muni/fi/DaoTests/ComponentDaoTest.java]: typo

Comment (RobertTamas) [reply src/main/test/cz/muni/fi/DaoTests/ComponentDaoTest.java]: :thumbsup:

File: src/main/java/cz/muni/fi/dao/CarDao.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
+package cz.muni.fi.dao;
+
+import cz.muni.fi.entities.Person;
```
Comment (matusmacko) [inline src/main/java/cz/muni/fi/dao/CarDao.java]: Isn't here needed `java.utils.list` import?

File: src/main/java/cz/muni/fi/dao/CarDao.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+/**
+ * Created by R. Tamas, 445241.
```
Comment (matusmacko) [inline src/main/java/cz/muni/fi/dao/CarDao.java]: Please, add @author tag here.

File: src/main/java/cz/muni/fi/dao/CarDao.java (hunk 2 above)
Comment (matusmacko) [inline src/main/java/cz/muni/fi/dao/CarDao.java]: Isn't here needed `java.utils.list` import?

File: src/main/java/cz/muni/fi/dao/CarDao.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+ */
+public interface CarDao {
+
+    /**
+     * List all cars which are in DB
+     * @return list of cars
```
Comment (matusmacko) [inline src/main/java/cz/muni/fi/dao/CarDao.java]: Please add space between description and other tags.

File: src/main/java/cz/muni/fi/dao/CarDao.java [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+     */
+    public List<Car> listAllCars();
+
+    /**
+     * Find specific car by its driver
+     * @param person driver
```
Comment (matusmacko) [inline src/main/java/cz/muni/fi/dao/CarDao.java]: Please add space between description and other tags.

File: src/main/test/cz/muni/fi/DaoTests/ComponentDaoTest.java [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.testng.AbstractTestNGSpringContextTests;
+
+import javax.persistence.EntityManager;
+import javax.persistence.EntityManagerFactory;
+import javax.persistence.PersistenceContext;
+import javax.persistence.PersistenceUnit;
+
+/**
+ * Created by R. Tamas, 445241.
```
Comment (matusmacko) [inline src/main/test/cz/muni/fi/DaoTests/ComponentDaoTest.java]: Please @author tag here

File: src/main/java/cz/muni/fi/enums/EDrivingSkill.java [hunk 7]
```diff
@@ -0,0 +1,33 @@
+package cz.muni.fi.enums;
+
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+
+/**
+ * Created by R. Tamas, 445241.
```
Comment (matusmacko) [inline src/main/java/cz/muni/fi/enums/EDrivingSkill.java]: Please @author tag here

File: src/main/java/cz/muni/fi/entities/Person.java [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+import java.util.EnumSet;
+import java.util.Set;
+
+import lombok.Getter;
+import lombok.Setter;
+
+import javax.persistence.*;
+
+/**
+ * Created by R. Tamas, 445241.
```
Comment (matusmacko) [inline src/main/java/cz/muni/fi/entities/Person.java]: Please @author tag here

File: src/main/java/cz/muni/fi/dao/CarDaoImpl.java [hunk 9]
```diff
@@ -0,0 +1,41 @@
+package cz.muni.fi.dao;
+
+import cz.muni.fi.entities.Person;
+import org.springframework.stereotype.Repository;
+
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
+
+/**
+ * Created by R. Tamas, 445241.
```
Comment (matusmacko) [inline src/main/java/cz/muni/fi/dao/CarDaoImpl.java]: Please @author tag here

File: src/main/java/cz/muni/fi/dao/CarDao.java [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+     * @return car with driver
+     */
+    public Car findCar(Person person);
+
+    /**
+     * Find specific car by its id
+     * @param id DB id
```
Comment (matusmacko) [inline src/main/java/cz/muni/fi/dao/CarDao.java]: Please add space between description and other tags.

File: src/main/java/cz/muni/fi/dao/CarDao.java [hunk 11]
```diff
@@ -0,0 +1,47 @@
...
+     * @return car
+     */
+    public Car findCar(Long id);
+
+    /**
+     * Adds car to DB
+     * @param car car to be added
```
Comment (matusmacko) [inline src/main/java/cz/muni/fi/dao/CarDao.java]: Please add space between description and other tags.

File: src/main/java/cz/muni/fi/dao/CarDao.java [hunk 12]
```diff
@@ -0,0 +1,47 @@
...
+     */
+    public void addCar(Car car);
+
+    /**
+     * Updates car in DB
+     * @param car car to be updated
```
Comment (matusmacko) [inline src/main/java/cz/muni/fi/dao/CarDao.java]: Please add space between description and other tags.

File: src/main/java/cz/muni/fi/dao/CarDao.java [hunk 13]
```diff
@@ -0,0 +1,47 @@
...
+     */
+    public void updateCar(Car car);
+
+    /**
+     * Deletes car from DB
+     * @param car car to be deleted
```
Comment (matusmacko) [inline src/main/java/cz/muni/fi/dao/CarDao.java]: Please add space between description and other tags.

File: src/main/java/cz/muni/fi/entities/Person.java [hunk 14]
```diff
@@ -0,0 +1,51 @@
...
+
+    @Getter private String name;
+    @Getter private String surname;
+    @Getter private String nationality;
+    @Getter boolean isMainDriver = false;
+
+    @Enumerated
+    @Getter private Set<EDrivingSkill> drivingSkills = EnumSet.allOf(EDrivingSkill.class);
+
+    public Person(String name, String surname, String nationality) {
```
Comment (matusmacko) [inline src/main/java/cz/muni/fi/entities/Person.java]: Pls add empty constructor

File: src/main/test/cz/muni/fi/DaoTests/ComponentDaoTest.java [hunk 15]
```diff
@@ -0,0 +1,92 @@
...
+    private EntityManager em;
+
+    @PersistenceUnit
+    private EntityManagerFactory emf;
+
+    @Autowired
+    private ComponentDaoImpl componentManager = new ComponentDaoImpl();
+
+
+    private Component engine1 = new Component();
```
Comment (matusmacko) [inline src/main/test/cz/muni/fi/DaoTests/ComponentDaoTest.java]: Pls move these properties to setup() method

File: src/main/java/cz/muni/fi/dao/CarDaoImpl.java [hunk 16]
```diff
@@ -0,0 +1,41 @@
...
+
+    @PersistenceContext
+    private EntityManager em;
+
+    public List<Car> listAllCars() {
+        return em.createQuery("select c from Car c", Car.class).getResultList();
+    }
+
+    public Car findCar(Person person) {
+        return em.createQuery("select c from Car c where c.person = :person", Car.class).setParameter("person", person).getResultList()[0];
```
[the remark this answers is not in the document]
Comment (RobertTamas) [reply src/main/java/cz/muni/fi/dao/CarDaoImpl.java]: Well i think it's meant like one driver can race with only one car at the same time.

[the remark this answers is not in the document]
Comment (matusmacko) [reply src/main/java/cz/muni/fi/dao/CarDaoImpl.java]: okay, so pls refactor as I wrote above

File: src/main/test/cz/muni/fi/DaoTests/ComponentDaoTest.java [hunk 17]
```diff
@@ -0,0 +1,92 @@
...
+        engine3.setType(EComponentType.ENGINE);
+        engine3.setIsAvailable(true);
+
+        em.persist(engine1);
+        em.persist(engine2);
+        em.persist(engine3);
+    }
+
+    @Test
+    public void listAvailableComponents() {
```
Comment (matusmacko) [inline src/main/test/cz/muni/fi/DaoTests/ComponentDaoTest.java]: Pls name test methods in this format listAvailableComponentsTest()

File: src/main/test/cz/muni/fi/DaoTests/ComponentDaoTest.java (hunk 15 above)
Comment (RobertTamas) [reply src/main/test/cz/muni/fi/DaoTests/ComponentDaoTest.java]: if i moved them i cannot access them from other methods then just `setup()`

Comment (matusmacko) [reply src/main/test/cz/muni/fi/DaoTests/ComponentDaoTest.java]: sure, checked only engine1 which is not used anywhere else.. i would move engine1 inside and rename the rest

File: src/main/java/cz/muni/fi/entities/Person.java [hunk 18]
```diff
@@ -0,0 +1,51 @@
...
+import java.util.Set;
+
+import lombok.Getter;
+import lombok.Setter;
+
+import javax.persistence.*;
+
+/**
+ * Created by R. Tamas, 445241.
+ */
```
Comment (matusmacko) [inline src/main/java/cz/muni/fi/entities/Person.java]: Pls add @Entity tag here

File: src/main/java/cz/muni/fi/entities/Person.java [hunk 19]
```diff
@@ -0,0 +1,51 @@
...
+public class Person {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    @Getter @Setter private Long id;
+
+    @Getter private String name;
```
Comment (matusmacko) [inline src/main/java/cz/muni/fi/entities/Person.java]: Pls add @NotNull tag where necessary

File: src/main/test/cz/muni/fi/DaoTests/ComponentDaoTest.java (hunk 17 above)
Comment (matusmacko) [reply src/main/test/cz/muni/fi/DaoTests/ComponentDaoTest.java]: And also move the test file outside of main dir.. wrote test dir structure on messenger

File: src/test/java/cz/muni/fi/DaoTests/ComponentDaoTest.java [hunk 20]
```diff
@@ -0,0 +1,97 @@
...
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.*;
+
+/**
+ * @author Robert Tamas
+ */
+
```
Comment (matusmacko) [inline src/test/java/cz/muni/fi/DaoTests/ComponentDaoTest.java]: Pls delete blank line here

File: src/test/java/cz/muni/fi/DaoTests/ComponentDaoTest.java [hunk 21]
```diff
@@ -0,0 +1,97 @@
...
+        aerodynamic.setType(EComponentType.AERODYNAMIC);
+        aerodynamic.setIsAvailable(true);
+
+        em.persist(transmission);
+        em.persist(engine);
+        em.persist(aerodynamic);
+    }
+
+    @Test
+    public void listAvailableComponents() {
```
Comment (matusmacko) [inline src/test/java/cz/muni/fi/DaoTests/ComponentDaoTest.java]: Pls change names to something like `listAvailableComponentsTest`

**Pull request merged**